### PR TITLE
docs(pr template): update pr template checklist to include translations FE-4144

### LINKED ISSUE #4109: Storybook : provide the resources for translation
### Current behaviour
In the storybook, each component has its list of props listed and explained.

### Desired behaviour
We would like the same thing for the potential resources used by that component.

For example, listing the following resources for the pager : pager.of_y, pager.records, .....
And the meaning of each.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 69.0.0 |

### PR DESCRIPTION
**Pull request by edleeks87**
fix #4109

### Proposed behaviour

Adds table of translation keys to storybook examples
Updates PR template checklist to include check for any translations added/ updated

### Current behaviour

Some components do not document available translation keys in storybook

### Checklist

- [x] Commits follow our style guide
- [x] Storybook added or updated if required

### Testing instructions

All stories for components mentioned in commit scope will need checking

### CODE REVIEW

**Review by nicktitchmarsh — DISMISSED** [review]

File: src/components/date/date.stories.mdx [hunk 1]
```diff
@@ -287,3 +289,27 @@ For more information check our [Validations](?path=/docs/documentation-validatio
...
+## Translation keys
+
+The following keys are available to override the translations for this component by passing in a custom locale object to the 
+[i18nProvider](https://carbon.sage.com/?path=/story/documentation-i18n--page).
+
+<TranslationKeysTable
+  translationData={
+    [
+      {
+        name: "date.formats.input",
```
Comment (mmaciorowski-sage) [inline src/components/date/date.stories.mdx:302]:
```suggestion
        name: "date.formats.inputs",
```

File: src/components/date-range/date-range.stories.mdx [hunk 2]
```diff
@@ -170,4 +172,28 @@ You can pass prop `allowEmptyValue` to start and end `DateInput` using `startDat
...
+## Translation keys
+
+The following keys are available to override the translations for this component by passing in a custom locale object to the 
+[i18nProvider](https://carbon.sage.com/?path=/story/documentation-i18n--page).
+
+<TranslationKeysTable
+  translationData={
+    [
+      {
+        name: "date.formats.input",
```
Comment (mmaciorowski-sage) [inline src/components/date-range/date-range.stories.mdx:186]:
```suggestion
        name: "date.formats.inputs",
```

**Comment by mmaciorowski-sage** [comment]
You probably skipped the files: 'step.js', 'pod.component.js', 'table.component.js', 'textarea.component.js'

**Comment by edleeks87** [comment]
> You probably skipped the files: 'step.js', 'pod.component.js', 'table.component.js', 'textarea.component.js'

@mmaciorowski-sage `step.js' and 'table.component.js' are marked for deprecation were intentionally skipped, whilst 'textarea.component.js only supports `locale()` rather than any specific translation keys and that is already documented on the `i18n.md` page so felt like overkill here. For `pod.component` I can see it uses `actions.edit` and I'll happily add it but the button/link will always just display with just an `Icon`, imo they should be updated to use `IconButton` and use an aria label really but that is out of scope here

**Review by nicktitchmarsh — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**